packages/agent-toolkit/src/core/tools/platform-api-tools/list-users-and-teams-tool/list-users-and-teams-tool.ts

```diff
@@ -0,0 +1,298 @@
+  name = 'list_users_and_teams';
+  type = ToolType.READ;
+  annotations = createMondayApiAnnotations({
+    title: 'List Users and Teams',
+    readOnlyHint: true,
+    destructiveHint: false,
+    idempotentHint: true,
+  });
+
+  getDescription(): string {
+    return `PRECISION-FIRST user and team retrieval tool. AI agents MUST prioritize specific queries over broad searches.
+
+      MANDATORY BEST PRACTICES:
+      1. ALWAYS use specific IDs when available (userIds, teamIds) - highest precision and performance
+      2. ALWAYS use name search when you have user names but no IDs  
+      3. ALWAYS use boolean getMe: true when requesting current user information
+      4. AVOID broad queries (no parameters) - use only as absolute last resort
+      5. COMBINE parameters strategically to minimize API calls
+
+      REQUIRED PARAMETER PRIORITY (use in this order):
+      1. getMe: true (when requesting current user) - STANDALONE ONLY
+      2. name="exact_name" (when searching by name) - STANDALONE ONLY  
+      3. userIds=["id1","id2"] (when you have specific user IDs)
+      4. teamIds=["id1","id2"] + teamsOnly: true (when you have specific team IDs)
+      5. No parameters (LAST RESORT - fetches up to 1000 users, avoid unless absolutely necessary)
+
+      CRITICAL USAGE RULES:
+      • getMe and name parameters CANNOT be combined with any other parameters
+      • userIds + teamIds requires explicit includeTeams: true flag
+      • teamsOnly: true prevents user data fetching (teams-only queries)
+      • includeTeamMembers: true adds detailed member data to teams
+
+      OPTIMIZATION DIRECTIVES:
+      • NEVER fetch all users when specific IDs are available
+      • NEVER use broad queries for single user/team lookups  
+      • ALWAYS prefer name search over ID-less queries for individual users
+      • SET includeTeamMembers: false for team lists, true only for member analysis  
+      • AVOID includeTeams: true unless you specifically need both users AND teams
+
+      RESPONSE CONTENT:
+      • Users: id, name, email, title, permissions, contact details, team memberships
+      • Teams: id, name, owners, members (when includeTeamMembers: true)
+      • Current user: id, name, title, enabled, is_admin, is_guest (basic profile only)`;
+  }
+
+  getInputSchema(): typeof listUsersAndTeamsToolSchema {
+    return listUsersAndTeamsToolSchema;
+  }
+
+  protected async executeInternal(
+    input: ToolInputType<typeof listUsersAndTeamsToolSchema>,
+  ): Promise<ToolOutputType<never>> {
+    const hasUserIds = input.userIds && input.userIds.length > 0;
+    const hasTeamIds = input.teamIds && input.teamIds.length > 0;
+    const includeTeams = input.includeTeams || false;
+    const teamsOnly = input.teamsOnly || false;
+    const includeTeamMembers = input.includeTeamMembers || false;
+    const hasName = !!input.name;
+    const getMe = input.getMe || false;
```

Comment:
In general here but for all parasm, instead of doing the || false you can assign default paramters in the ZOD schema